api/v1/slo_types.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// SloSpec defines the desired state of Slo
+type SloSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Foo is an example field of Slo. Edit slo_types.go to remove/update
+	Name        string            `json:"name"`
+	Description string            `json:"description,omitempty"`
+	Type        string            `json:"type"`
+	Window      string            `json:"window"`
+	Slo         int               `json:"slo"`
+	Threshold   int               `json:"threshold"`
+	Annotations map[string]string `json:"annotations,omitempty"`
+	Labels      map[string]string `json:"labels,omitempty"`
+}
+
+// SloStatus defines the observed state of Slo
+type SloStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	SloStatus string `json:"SloStatus"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// Slo is the Schema for the sloes API
+type Slo struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   []SloSpec `json:"spec,omitempty"`
```

Comment:
spec field as list ? sounds very strange (didn't saw it in any k8s resource) i think it's not right. let me know what you think.

in addition, i don't think the name should be sloSpec. maybe monitorSpec

Reply:
indeed no need, probably a leftover from an `init` flow in one of the tools.

Reply:
This is my solution to working with a structure similar to: 
```spec:
  - name: example-latency
    description: test
    type: latency
    window: 5m
    slo: 95
    threshold: 100
  - name: example-availability
    description: test
    type: latency
    window: 1h
    slo: 97
    threshold: 95
```
as we've discussed